src/ripple/rpc/handlers/ServerInfo.cpp

```diff
@@ -21,14 +21,377 @@
 #include <ripple/app/misc/NetworkOPs.h>
 #include <ripple/app/reporting/P2pProxy.h>
 #include <ripple/json/json_value.h>
+#include <ripple/json/json_writer.h>
 #include <ripple/net/RPCErr.h>
+#include <ripple/protocol/LedgerFormats.h>
+#include <ripple/protocol/digest.h>
+#include <ripple/protocol/SField.h>
 #include <ripple/protocol/jss.h>
 #include <ripple/rpc/Context.h>
 #include <ripple/rpc/Role.h>
 #include <ripple/rpc/impl/TransactionSign.h>
+#include <boost/algorithm/string.hpp>
+#include <magic/magic_enum.h>
+#include <sstream>
+
+#define MAGIC_ENUM(x)                           \
+    template <>                                 \
+    struct magic_enum::customize::enum_range<x> \
+    {                                           \
+        static constexpr int min = -20000;      \
+        static constexpr int max = 20000;       \
+    };
+
+MAGIC_ENUM(ripple::SerializedTypeID);
+MAGIC_ENUM(ripple::LedgerEntryType);
+MAGIC_ENUM(ripple::TELcodes);
+MAGIC_ENUM(ripple::TEMcodes);
+MAGIC_ENUM(ripple::TEFcodes);
+MAGIC_ENUM(ripple::TERcodes);
+MAGIC_ENUM(ripple::TEScodes);
+MAGIC_ENUM(ripple::TECcodes);
 
 namespace ripple {
 
+class Definitions
+{
+private:
+    Json::Value
+    generate()
+    {
+        // RH TODO: probably a better way to do this?
+#define STR(x)                      \
+    ([&] {                          \
+        std::ostringstream ss;      \
+        return ss << (x), ss.str(); \
+    }())
```

Comment:
Consider using a function instead. It'll fix compile error with apple `clang` (see comments below). Something like this should work.
```
template <typename T>
std::string
str(T const& t)
{
    std::ostringstream ss;
    ss << t;
    return ss.str();
}
```